src/bar.jl

```diff
@@ -7,12 +7,54 @@ end
 grouped_xy(x::AbstractVector, y::AbstractMatrix) = x, y
 grouped_xy(y::AbstractMatrix) = 1:size(y,1), y
 
+function homogenous_signs(y::AbstractMatrix)
+     for i in 1:size(y, 1)
+        sgn = 0.0
+        for j in 1:size(y,2)
+            testsgn = sign(y[i,j])
+            if sgn == 0.0
+                sgn = testsgn
+            end
+            if sgn != testsgn && testsgn != 0.0
+                return false
+            end
+        end
+    end
+    true
+end
+
+function groupedbar_fillrange(y::AbstractMatrix)
+    nr, nc = size(y)
+    y = copy(y)
+        y[.!isfinite.(y)] .= 0
+        fr = zeros(nr, nc)
+        for c=2:nc
+            for r=1:nr
+                fr[r,c] = y[r,c-1]
+                y[r,c] += fr[r,c]
+            end
+        end
+        y, fr
+end
+
+#=
```

Comment:
The package currently does not have tests, but it would be useful to add these to runtests.jl as:

```julia
@testset "groupedbar" begin
    @test homogenous_signs(rand(10,3))
    ...
end
```